utils/minpermwer.py

```diff
@@ -4,18 +4,47 @@
 # Copyright 2017 Johns Hopkins University (Shinji Watanabe)
 #           2018 Xuankai Chang (Shanghai Jiao Tong University)
 #  Apache 2.0  (http://www.apache.org/licenses/LICENSE-2.0)
-from __future__ import print_function
-from __future__ import unicode_literals
 
-import argparse
-import codecs
-import json
 import logging
-import numpy as np
 import re
 import sys
 
-is_python2 = sys.version_info[0] == 2
+
+# pre-set the permutation scheme (ref_idx, hyp_idx)
+def permutation_schemes(num_spkrs):
+    perm = [(i / num_spkrs + 1, i % num_spkrs + 1) for i in range(num_spkrs ** 2)]
+
+    # [r1h1, r1h2, r2h1, r2h2]
+    if num_spkrs == 2:
```

Comment:
Can we generalized this part given the number of speakers instead of writing the actual possible permutations? Then, you could generalize this script for any number of speakers.